src/WpfMath/Atoms/Atom.cs

```diff
@@ -1,19 +1,26 @@
+using System.Collections.Generic;
 using WpfMath.Boxes;
-
 namespace WpfMath.Atoms
 {
     // Atom (smallest unit) of TexFormula.
     internal abstract class Atom
     {
-        protected Atom(SourceSpan source, TexAtomType type = TexAtomType.Ordinary)
+        protected Atom(SourceSpan source, IEnumerable<Atom> children, TexAtomType type = TexAtomType.Ordinary)
         {
             this.Source = source;
             this.Type = type;
+            this.Children = children;
+        }
+
+        protected Atom(SourceSpan source, TexAtomType type = TexAtomType.Ordinary): this(source, new List<Atom>(), type)
+        {
         }
 
         public TexAtomType Type { get; }
 
-        public SourceSpan Source { get; }
+        public SourceSpan Source { get; set; }
+
+        public IEnumerable<Atom> Children { get; protected set; }
```

Comment:
Hello!

Excuse me.

But can we do without `IEnumerable` in public?

Reply:
Hello. 

I guess in the current implementation it isn't possible.

But we can something like that:

1. define an abstract method in Atom
`public abstract void MoveOffset(int position);`
2. Implement that in each child of Atom

Reply:
we have in stock  `protected` or `internal` (based on the situation) ;)

I have another questions now. Do you want to use the interface? We can get by with interfaces like `IReadOnlyList` or `IReadOnlyCollection` instead of `IEnumerable`?

Reply:
Yep, internal is fine.

I can use any of them. What do you recommend?

Reply:
> I can use any of them.

Fine :)

> What do you recommend?

less diff :)

There are no differences between your changes. It's just that one is the heir to the other and provides more options or restrictions, depending on which side you look. Just wanted to point out that using `IEnumerable` is not always good

Reply:
Our `Atom`s are already `internal`, so I can't see a particular point in changing the current visibility.

Wa have plans to potentially make them `public` later, and then we'll review all the available pubic API.

Also, I believe that `IEnumerable` is better here. Constructing and maintaining a `IReadOnlyList` would be more memory consuming IMO.

Reply:
Ok, if you think that for a property, in this situation, it is better to use `IEnumerable`, then so be it

Reply:
Yes, I think so. I would be glad to hear arguments why you think that's not the best idea though. We could easily reconsider if that's harmful in some way.

Reply:
I apologize, but my knowledge is bad, to understand why it is bad to use a collection in which we clearly know how many elements, or a collection in which we can access an element by index. I just go for general reasons that it is better and prevents unnecessary calculations in the future. I also walk out of considerations that right now this is actually `ReadOnlyCollection` anyway. On the other hand, I don’t remember the index call.